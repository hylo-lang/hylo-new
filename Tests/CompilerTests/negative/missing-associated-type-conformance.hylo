//! no-std

trait P {
  type X ; given X is P
  fun x() -> X
}

given {a: Void} is P {
  type X = {a: Void}
  public fun x() -> X { (a: ()) }
}

given {b: Void} is P {
  type X = {c: Void}
  public fun x() -> X { (c: ()) }
}

fun f<T where T is P>(_ t: T) {
  let x: T.X = t.x()
  f(x)
}
