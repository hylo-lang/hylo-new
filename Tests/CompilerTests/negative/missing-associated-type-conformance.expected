negative/missing-associated-type-conformance.hylo:13.1-6: error: no given instance of 'P<X>' in this scope
given {b: Void} is P {
~~~~~
